src/esphomelib/time/rtc_component.cpp

```diff
@@ -0,0 +1,58 @@
+//
+//  rtc.cpp
+//  esphomelib
+//
+//  Created by Brad Davidson on 06.06.18.
+//  Copyright © 2018 Otto Winter. All rights reserved.
+//
+
+#include "esphomelib/log.h"
+#include "esphomelib/time/rtc_component.h"
+
+#ifdef USE_RTC_COMPONENT
+
+ESPHOMELIB_NAMESPACE_BEGIN
+
+namespace time {
+
+static const char *TAG = "rtc";
+
+RTCComponent::RTCComponent(const std::string &tz){
+  this->set_timezone(tz);
+}
+void RTCComponent::setup(){
+}
+void RTCComponent::set_timezone(const std::string &tz){
+  this->tz_ = tz;
+  setenv("TZ", this->tz_.c_str(), 1);
+  tzset();
```

Comment:
This `setenv` and `tzset` code should go into a `setup()` method. esphomelib's API tries to pretty much everywhere it's possible store information in class members. And then only when `setup()` is executed it initializes the hardware.

```cp
RTCComponent::RTCComponent(const std::string &tz)
  : tz_(tz) {

}

void RTCComponent::setup() {
  setenv("TZ", this->tz_.c_str(), 1);
  tzset();
}
```